### PR TITLE
Add datatype inferral to convenience array

### PR DESCRIPTION
**Pull request by j6k4m8**
Will automatically redirect requests for cvdb data to the `intern.CloudVolumeRemote` with route information pre-filled.

### CODE REVIEW

File: intern/convenience/array.py [hunk 1]
```diff
@@ -222,6 +392,30 @@ def bulk_update(self, items: dict):
     def bulk_delete(self, keys: list):
         return self._remote.delete_metadata(self._resource, keys)
 
+
+def _infer_volume_provider(channel: Union[ChannelResource, str, Tuple]):
```
Comment (movestill) [inline intern/convenience/array.py:396]: When the channel is in a tuple, what does that look like?

Comment (j6k4m8) [reply intern/convenience/array.py:396]: At least internally, that means `[collection, experiment, channel]`, of type `Tuple[str, str, str]`. (We support that as a legacy option for passing them as `*args` to a Resource constructor)

**Review by sandyhider — DISMISSED** [review]

**Review by dxenes1 — DISMISSED** [review]
See comment about accessing cloudvolume channels though!

File: intern/convenience/array.py [hunk 2]
```diff
@@ -137,13 +185,117 @@ def create_cutout(
...
+    def get_vp_type(self) -> str:
+        return "CloudVolumeOpenData"
+
+    def get_axis_order(self) -> str:
+        return AxisOrder.XYZ
+
+    def get_channel(self, channel: str, collection: str, experiment: str):
+        return CloudVolumeResource(
+            self.cv_config["protocol"],
+            f"{self.cv_config['cloudpath']}/{collection}/{experiment}/{channel}",
```
[the remark this answers is not in the document]
Comment (j6k4m8) [reply intern/convenience/array.py]: Good call @dxenes1 — fixed!

**Review by dxenes1 — APPROVED** [review]

**Pull request merged**